src/backend/distributed/planner/multi_logical_optimizer.c

```diff
@@ -2907,14 +2907,15 @@ IsPartitionColumnRecursive(Expr *columnExpression, Query *query)
 	List *rangetableList = query->rtable;
 	Index rangeTableEntryIndex = 0;
 	RangeTblEntry *rangeTableEntry = NULL;
+	Expr *strippedColumnExpression = strip_implicit_coercions(columnExpression);
```

Comment:
Yeah, I think so. I'll update it to in my PR:
```C
Expr *strippedColumnExpression = strip_implicit_coercions((Node *) columnExpression);
```
But, I'm sure @anarazel will complain about why it is in fast shard pruning PR :))